utils.py

```diff
@@ -159,3 +159,12 @@ def _compare_endpoints(endpoint1, endpoint2):
     elif 'out_vlan' in endpoint1 or 'in_vlan' in endpoint2:
         return False
     return True
+
+
+def map_dl_vlan(vlan_value):
```

Comment:
A direct mapping won't suffice. Here with "4096/4096" we already have a concept of "value/mask" wildcard, and since `sdntrace_cp` cp is supposed to match based on the spec. Then you code need to at least take the current value being matched, and it should perform a logical or with the entry, for instance:

If you want to know if vlan=1 matches "4096/4096" you need to split the value and its 12-bit mask, and verify if vlan & (mask & 4095) == value & (mask & 4095) -> 4096 (mask) & 4096 will always be zero so any `vlan` & 0 will also be zero, which is why this is "any" it'll match any vlan. That way you can also reuse this part for vlan range when the time comes. 

![20230316_161441](https://user-images.githubusercontent.com/1010796/225730924-4ae13d5b-4340-4c9f-8c5e-a447336fccbf.png)

For more info check out the spec https://opennetworking.org/wp-content/uploads/2014/10/openflow-spec-v1.3.0.pdf. You can also check out how of_core serializes and deserialize this value https://github.com/kytos-ng/of_core/blob/master/v0x04/match_fields.py#L66-L73

Reply:
@viniarck I got it, thank you very much for the explanation, it has been a great help, now it is more clear.
Question, I could understand that `dl_vlan` = `untagged` is equivalent to the case where there is no `dl_vlan`. However, `untagged` is replaced by `0`, this means that `dl_vlan` = `0` must also be processed as missing field, so there will be no match and a trace with type `incomplete` will be obtained, right?

Reply:
You're welcome, Gretel. 

`dl_vlan: 0` can be thought of somewhat similar as when there's no `dl_vlan`, but it shouldn't be considered as a missing field, when `dl_vlan: 0` then it should only match the input arg if it also doesn't have a vlan set.